Code/GraphMol/QueryAtom.h

```diff
@@ -64,6 +64,19 @@ class RDKIT_GRAPHMOL_EXPORT QueryAtom : public Atom {
     }
     return *this;
   }
+
+  QueryAtom(QueryAtom &&other) noexcept : Atom(std::move(other)) {
+    dp_query = std::move(other.dp_query);
+  }
+  QueryAtom &operator=(QueryAtom &&other) noexcept {
+    if (this == &other) {
+      return *this;
+    }
+    QueryAtom::operator=(std::move(other));
+    dp_query = std::move(other.dp_query);
```

Comment:
Since dp_query is not of type class, we should use std::exchange

```
dp_query = std::exchange(other.dp_query, nullptr);
```

Reply:
Otherwise I think the original value is not zeroed.  (I admit I'm confused here, but it seems correct).